### DISCUSSION TITLE
Focus the current tab

### DISCUSSION THREAD

**Discussion by musjj** [wezterm/wezterm#3211] [Q&A] [answered]
I'm looking for a way to remotely send a command to neovim that would focus the Wezterm tab the neovim process is currently in.

nvim is capable of server-client communication, like this:
```bash
nvim --listen $ADDRESS
nvim --server $ADDRESS --remote-send "<C-\><C-N>:echo 'hello world'<CR>"
```

So if there's some kind of command that can somehow focus the current tab, we can do something like this:
```bash
nvim --server $ADDRESS --remote-send "<C-\><C-N>:!wezterm cli focus-this-tab<CR>" # fictional command
```

Is this too magical? Or is there a way to achieve this?

**Comment by yutkat** [comment]
I think it's impossible because wezterm cli doesn't have a subcommand to control tab focus.

https://wezfurlong.org/wezterm/cli/cli/index.html

**Comment by wez** [comment, marked as answer]
Take a look at https://github.com/wez/wezterm/issues/2979#issuecomment-1447519267 for a way to get this sort of thing working

**Reply by musjj** [reply]
Wow thanks, that's a really cool feature.

Is it possible to know the tab where the event was emitted from (so that we can focus it)?

**Reply by musjj** [reply]
Not sure if this is the "correct" method, but I've managed to figure it out:

```lua
wezterm.on("user-var-changed", function(window, pane, name, value)
  if name == "focus" then
    window:perform_action(wezterm.action.ActivateTab(pane:tab():tab_id()), pane)
    window:perform_action(wezterm.action.ActivatePaneByIndex(pane:pane_id()), pane)
    window:focus()
  end
end)
```
```bash
nvim --server $ADDRESS --remote-send "<C-\><C-N>:call chansend(v:stderr, \"\033]1337;SetUserVar=focus=\007\")<CR>"
```

Wezterm is really cool!
